unicore/hub/service/validators.py

```diff
@@ -1,25 +1,87 @@
+import re
+
 import colander
 
+from unicore.hub.service.models import App as AppModel, User as UserModel
+from unicore.hub.service.utils import translation_string_factory as _
+
+
+""" User schema validators
+"""
+
 
-def username_validator(node, value):
+# exclude control characters
+USERNAME_DISALLOWED_CHARS = u'\u0000-\u001f\u007f-\u009f'
+# have a look at http://unicode.org/reports/tr20/#Suitable
+# only excluding bidi controls
+USERNAME_DISALLOWED_CHARS += u'\u202a-\u202e'
+USERNAME_DISALLOWED_CHARS = re.compile(u'[%s]' % USERNAME_DISALLOWED_CHARS)
```

Comment:
Would there be any need for us to use `re.escape()` before feeding it to `re.compile()`?